### ISSUE TITLE
Quarkus 2.0.0 with resteasy-reactive and kotlin produces invalid OpenAPI schemas

### ISSUE THREAD

**Issue by nkhoshini** [quarkusio/quarkus#18468]
## Describe the bug
We are using Quarkus 2.0.0 internally and would like to switch to using the Kotlin coroutine support built into quarkus-resteasy-reactive. However the OpenAPI schema generated from the REST methods seem to add a coroutineContext value to the request.

A simple resource like
```
@Path("/hello")
class ReactiveGreetingResource {

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    suspend fun hello(): String {
        return "Hello RESTEasy Reactive"
    }
}
```

produces the following OpenAPI spec 

```
---
openapi: 3.0.3
info:
  title: Generated API
  version: "1.0"
paths:
  /hello:
    get:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/ContinuationSuperString'
      responses:
        "200":
          description: OK
          content:
            text/plain:
              schema:
                type: object
components:
  schemas:
    ContinuationSuperString:
      type: object
      properties:
        context:
          $ref: '#/components/schemas/CoroutineContext'
    CoroutineContext:
      type: object
```

Getting rid of the `suspend` modifier produces the proper OpenAPI result. 

### Expected behavior
To produce an OpenAPI spec without any coroutineContext info in the incoming payload

### Actual behavior

Each REST method that does not specify a request body has a `ContinuationSuper*` objects added by default as the request body value. 

## To Reproduce

1. Add resteasy-reactive, and openapi to your project
2. Create a suspending Kotlin resource method
3. Download/view the resultant OpenAPI yaml file

[code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/6773192/code-with-quarkus.zip)

### Screenshots
(If applicable, add screenshots to help explain your problem.)

### Output of `uname -a` or `ver`
Darwin 6.local 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64

### Output of `java -version`
java version "11.0.6" 2020-01-14 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.6+8-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.6+8-LTS, mixed mode)

### Quarkus version or git rev
2.0.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Gradle 6.8.3

**Comment by phillip-kruger** [comment]
I created https://github.com/smallrye/smallrye-open-api/issues/848 for this. We need to fix this in SmallRye

**Comment by geoand** [comment]
Thanks @phillip-kruger

**Comment by MikeEdgar** [comment]
@nkhoshini - I am not getting the same results with the attached example project. Do you have `javax.ws.rs.Consumes` annotation somewhere in your project that is not in the example? I ask because the annotation scanner does not consider a method argument (in this case a `kotlin.coroutines.Continuation`) to be the request body without it.

**Comment by MikeEdgar** [comment]
PR smallrye/smallrye-open-api#883 opened upstream for this. Anyone interested in this issue and more familiar with Kotlin, please feel free to chime in and review the proposed changes!

**Comment by cogman** [comment]
I see that the smallrye-open-api change should be in as part of Quarkus 2.2.1,  However I'm still seeing this issue at that version.

Further, another quirk (Not sure if it's related) is that the request body on @GET requests is being set to the surrounding resource class.

For example

    get:
      tags:
      - Foo
      operationId: getFoo
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/FooResource'

**Comment by MikeEdgar** [comment]
@cogman - would you mind opening an issue in [SmallRye OpenAPI](https://github.com/smallrye/smallrye-open-api) with sample Kotlin code?

**Comment by geoand** [comment]
Is this still an issue?

**Comment by phillip-kruger** [comment]
As far as I know this is fixed

**Comment by geoand** [comment]
Thanks, let's close then

**Comment by MikeEdgar** [comment]
@vladimirfx - would you mind to open an issue in [smallrye/smallrye-openapi](https://github.com/smallrye/smallrye-open-api/issues) with an reproduce app linked/attached?

**Comment by vladimirfx** [comment]
Fixed in 2.5.1.Final
